newIDE/app/src/EventsFunctionsExtensionsLoader/index.js

```diff
@@ -350,20 +352,16 @@ const generateFreeFunction = (
   eventsFunction: gdEventsFunction,
   options: OptionsForGeneration,
   codeGenerationContext: CodeGenerationContext
-): Promise<{
-  functionFile: string,
-  functionMetadata:
-    | gdInstructionMetadata
-    | gdExpressionMetadata
-    | gdMultipleInstructionMetadata,
-}> => {
-  const { functionFile, functionMetadata } = generateFreeFunctionMetadata(
+): Promise<void> => {
+  const metadataDeclarationHelper = new gd.MetadataDeclarationHelper();
+  const { functionMetadata } = generateFreeFunctionMetadata(
     project,
     extension,
     eventsFunctionsExtension,
     eventsFunction,
     options,
-    codeGenerationContext
+    codeGenerationContext,
+    metadataDeclarationHelper
   );
```

Comment:
`generateFreeFunctionMetadata` is not static because it return a `AbstractFunctionMetadata` which can be a `MultipleInstructionMetadata`.